app/services/appropriate_bodies/process_batch/action.rb

```diff
@@ -0,0 +1,161 @@
+module AppropriateBodies
+  module ProcessBatch
+    # Management of induction periods in bulk via CSV upload
+    # Handles Pass / Fail / Release in two stages
+    class Action < Base
```

Comment:
This class is doing a lot, for example: 

- `fetch_trs_details`
- `validate_submission!`
- `do_action`

Might be worth exploring whether any of these could be pulled out into seperate service objects or classes to try and abide by the single responsibility principle a bit more.

What do you think?

Reply:
That's certainly the direction I would expect. `Claim` reuses existing services for example, and here I originally tried to, but it was awkward and I really wanted to avoid making changes that might alter existing services. It's giving `monad`. Some of the CSV-level validations are moving to row-level, so the logic of the validation section is about to grow. Also I'd agree that we should avoid multiple places fetching from TRS. I am working on refactoring the validations in a separate branch and will make splitting the service part of its focus.